routers/action-router.js

```diff
@@ -0,0 +1,106 @@
+const express = require('express');
+const db = require('../data/helpers/projectModel.js');
+const actionDb = require('../data/helpers/actionModel.js');
+
+const router = express.Router();
+
+router.get('/:id/actions', (req, res) => {
+    db.getProjectActions(req.params.id)
+        .then((actions) => {
+            res.status(200).json(actions)
+        })
+        .catch(() => {
+            res.status(500).json({message: "There was a problem retrieving the actions."})
+        })
+})
+
+router.get('/:id/actions/:actionId', validateProjectId, validateActionId, (req, res) => {
+    const {id, actionId} = req.params;
+
+    actionDb.get(actionId)
+        .then((action) => {
+            res.status(200).json(action);
+        })
+        .catch(() => {
+            res.status(500).json({message: "There was a problem retrieving the action."})
+        })
+})
+
+router.post('/:id/actions', validateProjectId, (req, res) => {
+    
+    const actionInfo = req.body;
+    
+    actionDb.insert(actionInfo)
+        .then((action) => {
+            res.status(201).json(action);
+        })
+        .catch(() => {
+            res.status(500).json({ message: "Error creating new action" });
+        });
+});
+
+router.put('/:id/actions/:actionId', validateProjectId, validateActionId, (req, res) => {
+    const updatedAction = req.body;
+
+    actionDb.update(req.params.actionId, updatedAction)
+      .then(() => {
+        actionDb.get(req.params.actionId)
+          .then(action => {
+            res.status(201).json(action);
```

Comment:
Also, you are referencing actionId through req.parms again here so you should definitely be pulling it out according to DRY.